src/web/lib/useHasBeenSeen.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState, useRef } from 'react';
+
+const useHasBeenSeen = (options: IntersectionObserverInit) => {
+    const [hasBeenSeen, setHasBeenSeen] = useState<boolean>(false);
+    const [node, setNode] = useState<HTMLElement | null>(null);
+
+    const observer = useRef<IntersectionObserver | null>(null);
+
+    useEffect(() => {
+        if (observer.current) observer.current.disconnect();
+
+        observer.current = new window.IntersectionObserver(([entry]) => {
+            if (entry.isIntersecting) setHasBeenSeen(true);
+        }, options);
+
+        const { current: currentObserver } = observer;
+
+        if (node) currentObserver.observe(node);
+
+        return () => currentObserver.disconnect();
```

Comment:
```suggestion
        if (observer.current) {
            observer.current.disconnect();
        }

        observer.current = new window.IntersectionObserver(([entry]) => {
            if (entry.isIntersecting) {
                setHasBeenSeen(true);
            }
        }, options);

        const { current: currentObserver } = observer;

        if (node) {
            currentObserver.observe(node);
        }

        return () => currentObserver.disconnect();
```

Reply:
sorry, just a comprehension/preference thing :/ I'm surprised we don't lint it

Reply:
Good point! Yeah I'd be in favour of linting on this if it's the project style. We do the same in the contributions service [here](https://github.com/guardian/contributions-service/blob/master/.eslintrc.js#L18)